src/pages/GetStarted/CustomWorkspaceTab/DevfileSelector.tsx

```diff
@@ -57,41 +63,52 @@ export class DevfileSelectorFormGroup extends React.PureComponent<Props, State>
     this.devfileLocationRef = React.createRef();
   }
 
-  private onDevfileSelected(meta: che.DevfileMetaData): void {
+  private onDevfileClear(): void {
+    if (this.props.onClear) {
+      this.props.onClear();
+    }
+  }
+
+  private async onDevfileSelected(meta: che.DevfileMetaData): Promise<void> {
     // clear location input
     this.devfileLocationRef.current?.clearInput();
-
     try {
-      this.loadDevfile(meta.links.self);
-    } catch {
-      // noop
+      const devfileContent = await this.props.requestDevfile(meta.links.self);
+      const devfile = safeLoad(devfileContent);
+      this.props.onDevfile(devfile);
+    } catch (e) {
+      let errorMessage = 'Failed to load devfile.';
+      console.warn(errorMessage, e);
+      if (typeof e === 'object' && e.toString) {
```

Comment:
`e` is always of `Error` type, you don't have to check it for the `toString`.

Reply:
done